prototyping/xarray/xdose.py

```diff
@@ -0,0 +1,198 @@
+        or np.array_equal(np.abs(orientation), np.array([0, 1, 0, 1, 0, 0]))
+    ):
+        raise ValueError(
+            "Dose grid orientation is not supported. Dose "
+            "grid slices must be aligned along the "
+            "superoinferior axis of patient."
+        )
+
+    is_decubitus = dataset_orientation_is_decubitus(ds)
+    is_head_first = dataset_orientation_is_head_first(ds)
+
+    di = float(ds.PixelSpacing[0])
+    dj = float(ds.PixelSpacing[1])
+
+    col_range = np.arange(0, ds.Columns * di, di)
+    row_range = np.arange(0, ds.Rows * dj, dj)
+
+    if is_decubitus:
+        x_support = orientation[1] * position[1] + col_range  # type: ignore
+        z_support = -(orientation[3] * position[0] + row_range)  # type: ignore
+    else:
+        x_support = orientation[0] * position[0] + col_range  # type: ignore
+        z_support = -(orientation[4] * position[1] + row_range)  # type: ignore
+
+    if is_head_first:
+        y_support = position[2] + np.array(ds.GridFrameOffsetVector)
+    else:
+        y_support = -position[2] + np.array(ds.GridFrameOffsetVector)  # type: ignore
+
+    if coord_system.upper() in ("SUPPORT", "IEC SUPPORT", "S"):
+        return (x_support, y_support, z_support)
+
+    elif coord_system.upper() in ("DICOM", "D", "PATIENT", "IEC PATIENT", "P"):
+
+        x_patient = orientation[0] * x_support + orientation[3] * z_support
+        z_patient = orientation[1] * x_support + orientation[4] * z_support
+
+        if not is_head_first:
+            y_patient = -y_support
+        else:
+            y_patient = y_support
+
+        if coord_system.upper() in ("PATIENT", "IEC PATIENT", "P"):
+            return (x_patient, y_patient, z_patient)
+
+        elif coord_system.upper() in ("DICOM", "D"):
+
+            x_dicom = x_patient
+            y_dicom = -z_patient
+            z_dicom = y_patient
+
+            return (x_dicom, y_dicom, z_dicom)
+
+
+def xarray_dims_from_dataset(ds, coord_system):
+
+    if coord_system.upper() in ("SUPPORT", "IEC SUPPORT", "S"):
+        return XARRAY_DIMS_BY_COORD_SYSTEM_AND_ORIENT["S"]
+
+    elif coord_system.upper() in ("PATIENT", "IEC PATIENT", "P"):
+        if dataset_orientation_is_decubitus(ds):
+            return XARRAY_DIMS_BY_COORD_SYSTEM_AND_ORIENT["P dec"]
+        else:
+            return XARRAY_DIMS_BY_COORD_SYSTEM_AND_ORIENT["P"]
+
+    elif coord_system.upper() in ("DICOM", "D"):
+        if dataset_orientation_is_decubitus(ds):
+            return XARRAY_DIMS_BY_COORD_SYSTEM_AND_ORIENT["D dec"]
+        else:
+            return XARRAY_DIMS_BY_COORD_SYSTEM_AND_ORIENT["D"]
+
+
+def xarray_coords_from_dataset(
+    ds, coord_system: str = "SUPPORT", round_to_decimals: int = -1
+):
+    x, y, z = coords_from_dataset(ds, coord_system=coord_system)
+
+    xcoords = {"x": x, "y": y, "z": z}
+
+    if round_to_decimals >= 0:
+        return {
+            dim: np.round(xc, decimals=round_to_decimals) for dim, xc in xcoords.items()
+        }
```

Comment:
See comment above, I think rounding here is making this function step outside its purview.